src/states/Lobby.gd

```diff
@@ -91,11 +101,33 @@ func on_start_button_press() -> void:
 remote func start_game() -> void:
 	for player_id in chosen_colors.keys():
 		Multiplayer.player_info[player_id].color = COLORS[chosen_colors[player_id]]
+		Multiplayer.player_info[player_id].team_id = chosen_colors[player_id]
 	var error := get_tree().change_scene("res://src/states/Game.tscn")
 	assert(not error)
 
 
-# Called when we press a button.
+# Called whenever someone selects a mode from the drop-down.
+# :param new_mode_id: The ID of the selected mode.
+func on_mode_select(new_mode_id: int) -> void:
+	# If we select the same game mode we have already selected, do nothing.
+	if new_mode_id == Multiplayer.game_mode:
+		return
+	sync_mode(new_mode_id)
+	rpc("sync_mode", Multiplayer.game_mode)
+
+
+# Set the game mode to the specified value.
+# :param new_mode_id: The ID of the mode to set.
+remote func sync_mode(new_mode_id: int) -> void:
+	Multiplayer.game_mode = new_mode_id
+	mode_drop_down.text = mode_drop_down.get_popup().get_item_text(new_mode_id)
+	# Because the mode changed, color selections are probably no longer valid. Reset them.
+	chosen_colors = {}
```

Comment:
We might not need to actually do this - I remember discussing that we might make it so everyone can just choose their colors regardless of everyone else and the match can only start if the color combination is valid. This solution is valid too but we might wanna discuss a bit more

Reply:
Perhaps. I like the way the disabled/focused buttons looked and how they indicated which color you selected. Now that I think about it though, if we removed that logic but just made a big ass circle in the middle with your color choice it could be better.